### PR TITLE
Filter impact function

### LINKED ISSUE #330: `filter_impact_function` is deprecated
child of #53

we need to remove every impact function related stuff in the code

### PR DESCRIPTION
**Pull request by myarjunar**
fix #330  - don't merge this before #329

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -124,6 +138,13 @@ class Meta:
         blank=False,
         null=False
     )
+    impact_function_name = models.CharField(
```
Comment (lucernae) [inline models.py:141]: We don't need this anymore

File: models.py [hunk 2]
```diff
@@ -256,16 +277,10 @@ def get_default_impact_title(self):
     @classmethod
     def impact_function_list(cls):
         if not cls._impact_function_list:
-            from geosafe.tasks.headless.analysis import filter_impact_function
-            cls._impact_function_list = filter_impact_function.delay().get()
+            from geosafe.tasks.analysis import filter_impact_function
```
Comment (lucernae) [inline models.py:280]: We can delete this function

File: tasks/analysis.py [hunk 3]
```diff
@@ -240,6 +236,40 @@ def clean_impact_result():
             i.delete()
 
 
+def filter_impact_function(hazard=None, exposure=None):
```
Comment (lucernae) [inline tasks/analysis.py:239]: We don't need this @myarjunar

File: views/analysis.py [hunk 4]
```diff
@@ -355,15 +356,7 @@ def impact_function_filter(request):
...
-        # need to update function
-
-        # async_result = filter_impact_function.delay(
-        #     hazard_url,
-        #     exposure_url)
-        #
-        # impact_functions = async_result.get()
-
-        impact_functions = []
+        impact_functions = filter_impact_function(hazard_url, exposure_url)
```
Comment (lucernae) [inline views/analysis.py:359]: We can also delete this.

**Review by lucernae — COMMENTED** [review]
Thanks @myarjunar, however note that we don't need to provide impact function list in InaSAFE v4. We just assume it is going to work. (If it's not, it's going to be reported from analysis result).

Maybe we can close this?

**Comment by myarjunar** [comment]
thanks @lucernae , I will close this and create another PR to replace every IF related stuff in the code

**Pull request closed without merging**